src/System.Drawing.Common/tests/FontTests.cs

```diff
@@ -19,5 +301,575 @@ public void Ctor_NoSuchFamilyName_SetsFamilyToGenericSansSerif(string familyName
+
+                var logFont = new LOGFONT();
+                font.ToLogFont(logFont, graphics);
+
+                Assert.Equal(-13, logFont.lfHeight);
+                Assert.Equal(0, logFont.lfWidth);
+                Assert.Equal(0, logFont.lfEscapement);
+                Assert.Equal(0, logFont.lfOrientation);
+                Assert.Equal(400, logFont.lfWeight);
+                Assert.Equal(0, logFont.lfItalic);
+                Assert.Equal(0, logFont.lfUnderline);
+                Assert.Equal(0, logFont.lfStrikeOut);
+                Assert.Equal(1, logFont.lfCharSet);
+                Assert.Equal(0, logFont.lfOutPrecision);
+                Assert.Equal(0, logFont.lfClipPrecision);
+                Assert.Equal(0, logFont.lfQuality);
+                Assert.Equal(0, logFont.lfPitchAndFamily);
+                Assert.Equal(family.Name, logFont.lfFaceName);
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Throws an AccessViolationException in the .NET Framework.")]
+        public void ToLogFont_NullLogFont_ThrowsArgumentNullException()
+        {
+            using (FontFamily family = FontFamily.GenericMonospace)
+            using (var font = new Font(family, 10))
+            using (var image = new Bitmap(10, 10))
+            using (Graphics graphics = Graphics.FromImage(image))
+            {
+                AssertExtensions.Throws<ArgumentNullException>("ptr", () => font.ToLogFont(null));
+                AssertExtensions.Throws<ArgumentNullException>("ptr", () => font.ToLogFont(null, graphics));
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ToLogFont_NullGraphics_ThrowsArgumentNullException()
+        {
+            using (FontFamily family = FontFamily.GenericMonospace)
+            using (var font = new Font(family, 10))
+            {
+                AssertExtensions.Throws<ArgumentNullException>("graphics", () => font.ToLogFont(new LOGFONT(), null));
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ToLogFont_DisposedGraphics_ThrowsArgumentException()
+        {
+            using (FontFamily family = FontFamily.GenericMonospace)
+            using (var font = new Font(family, 10))
+            using (var image = new Bitmap(10, 10))
+            {
+                Graphics graphics = Graphics.FromImage(image);
+                graphics.Dispose();
+
+                AssertExtensions.Throws<ArgumentException>(null, () => font.ToLogFont(new LOGFONT(), graphics));
+            }
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+        public class LOGFONT
```

Comment:
I have a feeling this is going to be problematic when we start running against libgdiplus, but we can deal with that when it occurs.